base/inference.jl

```diff
@@ -1135,7 +1099,7 @@ function abstract_eval(e::ANY, vtypes, sv::StaticVarInfo)
             end
         end
     elseif is(e.head,:method)
-        t = Function
+        t = (length(e.args) == 1) ? Any : Void
```

Comment:
The 4-argument form of `:method` only mutates a method table, and so returns nothing.